### PR TITLE
attempt to retry snap commands using ts-retry

### PR DESCRIPTION
**Pull request by addyess**
include [`ts-retry`](https://www.npmjs.com/package/ts-retry) to try out some very basic retry logic on the snap and apt-get commands

### CODE REVIEW

File: src/bootstrap/index.ts [hunk 1]
```diff
@@ -110,6 +111,13 @@ async function microk8s_init(addons) {
 }
 
 
+const _retryable_snap = async (snap_args: string, args?: string[], options?: exec.ExecOptions): Promise<number> => {
+    // Run a snap command yielding the awaited Promise result
+    return await exec.exec(`sudo snap ${snap_args}`, args, options);
+};
+const snap = retryAsyncDecorator(_retryable_snap, {delay:100, maxTry:5 });
```
Comment (addyess) [inline src/bootstrap/index.ts:118]: obviously hard-coded the library to retry 5 days, 100ms apart -- any failed Promise should be retried

Comment (ca-scribner) [reply src/bootstrap/index.ts:118]: *5 times.  5 days would be a bit much :)

File: src/bootstrap/index.ts [hunk 2]
```diff
@@ -110,6 +112,20 @@ async function microk8s_init(addons) {
 }
 
 
+const _retryable_exec = (command: string, maxTry: number = 5) => {
+    // returns an async method capable of running the prog with sudo
+    const fn = async (cmd_arg:string, args?: string[], options?: exec.ExecOptions): Promise<number> => {
+        // Run a command with sudo yielding the awaited Promise result
+        return await exec.exec(`sudo ${command} ${cmd_arg}`, args, options);
+    };
+    const backoff = createExponetialDelay(100);
```
Comment (addyess) [inline src/bootstrap/index.ts:121]: i'm aware of the misspelling of `exponential` here -- but it's from the upstream library

**Review by ca-scribner — APPROVED** [review]
LGTM!

During testing today due to intermittent outages with the `snap` servers, we could confirm that this properly handled lost connections during `snap` calls.  We tried to add tests using an artificially flaky version of the `snap` command, but that broke `lxd waitready` calls as well so we pulled those tests.

**Pull request merged**